src/components/SearchToolbar/CategoryButton.js

```diff
@@ -44,24 +45,40 @@ function CategoryButton(props: Props) {
   );
 
   return (
-    <Button
-      onClick={() => props.onClick(category)}
-      className={className}
-      onFocus={props.onFocus}
-      aria-label={showCloseButton ? t`Remove ${props.name} Filter` : props.name}
-    >
-      <IconButton isHorizontal={showCloseButton} caption={props.name} hasCircle={props.hasCircle}>
-        {icon}
-      </IconButton>
-      {showCloseButton && <CloseIcon />}
-    </Button>
+    <RouteConsumer>
+      {context => {
+        const params = { ...context.params };
+
+        if (showCloseButton) {
+          delete params.category;
+        } else {
+          params.category = category;
```

Comment:
0: Totally unrelated, but it's confusing that `this.props.category` is not a Category but a categoryId

Reply:
This is not a property but a param. But you're right! Would fix this in a separate task/PR.